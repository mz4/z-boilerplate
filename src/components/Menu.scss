.dropdown-content {
  display: block;
  position: absolute;
  background: #f1f1f1;
  min-width: 160px;
  overflow: auto;
  z-index: 1;
}

.dropdown-content a {
  color: black;
  padding: 12px 16px;
  text-decoration: none;
  display: block;
}
