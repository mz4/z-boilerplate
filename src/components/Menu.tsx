import React from "react";
import "./Menu.scss";

interface IProps {
  isMenuOpen: boolean;
}

const Menu = ({ isMenuOpen }: IProps) => {
  return (
    <div className="dropdown-content">
      {isMenuOpen ? (
        <>
          <a href="#home">Home</a>
          <a href="#about">About</a>
          <a href="#contact">Contact</a>
        </>
      ) : null}
    </div>
  );
};

export default Menu;
