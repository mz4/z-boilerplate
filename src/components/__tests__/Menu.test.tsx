import React from "react";
import { render } from "@testing-library/react";
import Menu from "../Menu";

const propertiesV1 = {
  isMenuOpen: false
}

const propertiesV2 = {
  isMenuOpen: true
}

describe('Menu', () => {
  it('should match snapshot', () => {
    const { container } = render(<Menu {...propertiesV1} />);
    expect(container).toMatchSnapshot();
  });
  it('should have menu close', () => {
    const { queryByText } = render(<Menu {...propertiesV1} />);
    expect(queryByText('Home')).not.toBeInTheDocument();
    expect(queryByText('About')).not.toBeInTheDocument();
    expect(queryByText('Contact')).not.toBeInTheDocument();
  });
});

describe('Menu', () => {
  it('should match snapshot', () => {
    const { container } = render(<Menu {...propertiesV2} />);
    expect(container).toMatchSnapshot();
  });
  it('should have menu open', () => {
    const { queryByText } = render(<Menu {...propertiesV2} />);
    expect(queryByText('Home')).toBeInTheDocument();
    expect(queryByText('About')).toBeInTheDocument();
    expect(queryByText('Contact')).toBeInTheDocument();
  });
});
