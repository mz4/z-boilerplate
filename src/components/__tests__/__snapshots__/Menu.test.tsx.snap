// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should match snapshot 1`] = `
<div>
  <div
    class="dropdown-content"
  />
</div>
`;

exports[`Menu should match snapshot 2`] = `
<div>
  <div
    class="dropdown-content"
  >
    <a
      href="#home"
    >
      Home
    </a>
    <a
      href="#about"
    >
      About
    </a>
    <a
      href="#contact"
    >
      Contact
    </a>
  </div>
</div>
`;
